salt/modules/groupadd.py

```diff
@@ -123,41 +165,74 @@ def getent(refresh=False):
         return __context__['group.getent']
 
     ret = []
-    for grinfo in grp.getgrall():
+    if root is not None:
+        _fun = functools.partial(_getgrall, root=root)
+    else:
+        _fun = functools.partial(grp.getgrall)
+
+    for grinfo in _fun():
         ret.append(_format_info(grinfo))
     __context__['group.getent'] = ret
     return ret
 
 
+def _chattrib(name, key, value, param, root=None):
+    '''
+    Change an attribute for a named user
+    '''
+    pre_info = info(name, root=root)
+    if not pre_info:
+        return False
+
+    if value == pre_info[key]:
+        return True
+
+    cmd = ['groupmod']
+
+    if root is not None:
+        cmd.extend(('-R', root))
+
+    cmd.extend((param, value, name))
+
+    __salt__['cmd.run'](cmd, python_shell=False)
```

Comment:
Some checks/logging would help here for debugging.

Reply:
Indeed, can we do that in a different change? The original code do not provide any check or log here for useradd, groupadd and shadow. I this this change for avoid code duplication, but I take care or not changing the semantic.

Adding some checks here can be normalized in a different commit, and this will require adjusting the tests.